### PR TITLE
Add sample to update protection run snapshot retention

### PR DESCRIPTION
**Pull request by ashish-cohesity**

### CODE REVIEW

File: samples/update_protection_runs/README.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+First make sure that you are connected to a Cohesity Cluster.
+```
+cohesity_client = CohesityClient(cluster_vip=CLUSTER_VIP,
+                                 username=CLUSTER_USERNAME, 
+                                 password=CLUSTER_PASSWORD,
+                                 domain=DOMAIN)
+```
+## Example
+``` 
+ body.job_runs.append(UpdateProtectionJobRun())
```
Comment (ggCohesity) [inline samples/update_protection_runs/README.md]: We can have a short but complete example function, to show implementation.

File: samples/update_protection_runs/update_job_run.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
+# Copyright 2019 Cohesity Inc.
+#
+# Python example to extend the snapshot retention per protection job run. The
+# runs can be filtered by start/end date.
+#
+# Usage : python update_job_run.py --job_name=<protect-job>
+# --start_time=2019-2-31 --end_time=2019-7-29 --extend=<num-of-days>
```
Comment (ggCohesity) [inline samples/update_protection_runs/update_job_run.py]: --start_time must be --start_date same with --end_time.

File: samples/update_protection_runs/update_job_run.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+                                                             'Snapshot?'))
+    for run in job_run:
+        print ('{0:<10}\t\t{1:>10}\t\t{2:>10}\t\t{3:>10}'.format(
+            str(convert_to_date(run.backup_run.stats.start_time_usecs)),
+            run.backup_run.status,
+            str(job_run[0].backup_run.stats.num_successful_tasks) + '/' +
+            str(job_run[0].backup_run.stats.num_failed_tasks), 'Yes'))
+
+def main(args):
+
```
Comment (ggCohesity) [inline samples/update_protection_runs/update_job_run.py]: Need to have a try-except block for exception handling..

File: samples/update_protection_runs/update_job_run.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    :param cluster_id:
+    :param incarnation_id:
+    :param extend:
+    :return:
+    """
+    body = UpdateProtectionJobRunsParam()
+    body.job_runs = []
+    index = 0
+
+    for run in job_run:
```
Comment (ggCohesity) [inline samples/update_protection_runs/update_job_run.py]: Check for the empty response for job_run

File: samples/update_protection_runs/update_job_run.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        body.job_runs[index].job_uid.cluster_incarnation_id = incarnation_id
+        body.job_runs[index].job_uid.id = job_id
+        body.job_runs[index].run_start_time_usecs = run.copy_run[0].run_start_time_usecs
+        body.job_runs[index].copy_run_targets = []
+        body.job_runs[index].copy_run_targets.append(RunJobSnapshotTarget())
+        body.job_runs[index].copy_run_targets[0].days_to_keep = int(extend)
+        body.job_runs[index].copy_run_targets[0].mtype = \
+            TypeRunJobSnapshotTargetEnum.KLOCAL
+        index += 1
+    cohesity_client.protection_runs.update_protection_runs(body=body)
```
Comment (ggCohesity) [inline samples/update_protection_runs/update_job_run.py]: Need Exception Handling for update API fail.

File: samples/update_protection_runs/update_job_run.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+            run.backup_run.status,
+            str(job_run[0].backup_run.stats.num_successful_tasks) + '/' +
+            str(job_run[0].backup_run.stats.num_failed_tasks), 'Yes'))
+
+def main(args):
+
+    start_time = convert_to_epoch(args.start_time.split('-')) + MILLISECONDS
+    end_time = convert_to_epoch(args.end_time.split('-'), True) + MILLISECONDS
+
+    job_id, cluster_id, incarnation_id = get_pro_id(args.job_name)
```
Comment (ggCohesity) [inline samples/update_protection_runs/update_job_run.py]: Exception handling for get_pro_id empty response

File: samples/update_protection_runs/README.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+ body.job_runs[index].copy_run_targets[0].days_to_keep = int(extend)
+ body.job_runs[index].copy_run_targets[0].mtype = \
+            TypeRunJobSnapshotTargetEnum.KLOCAL
+ index += 1
+ cohesity_client.protection_runs.update_protection_runs(body=body)
+
+```
+## Example Output
+```
+Start Time		                  Status		Success/Error		Extended Snapshot?
```
Comment (ggCohesity) [inline samples/update_protection_runs/README.md]: Can we show previous values along with updated values?

File: samples/update_protection_runs/README.md (hunk 1 above)
Comment (ashish-cohesity) [reply samples/update_protection_runs/README.md]: Hmm, readme should have actual body/payload, this is just to copy paste the code if the customers going through the examples are in doubt.

File: samples/update_protection_runs/update_job_run.py (hunk 2 above)
Comment (ashish-cohesity) [reply samples/update_protection_runs/update_job_run.py]: Ack, good catch.

File: samples/update_protection_runs/README.md (hunk 7 above)
Comment (ashish-cohesity) [reply samples/update_protection_runs/README.md]: Good suggestion, but I think It's ok , consistent with our UI .

File: samples/update_protection_runs/update_job_run.py (hunk 4 above)
Comment (ashish-cohesity) [reply samples/update_protection_runs/update_job_run.py]: Good catch!

File: samples/update_protection_runs/update_job_run.py (hunk 6 above)
Comment (ashish-cohesity) [reply samples/update_protection_runs/update_job_run.py]: handled in the function.

**Pull request merged**